fixing lombok constants

sebing opened the pull request:
Fixing the issue [issues/697](https://github.com/yegor256/takes/issues/697)

I have removed the parameters for 

```
@ToString
@EqualsAndHashCode
```

As "forks" is the only field, and by default all the fields should be included in 
- The hashing calculation
- Equals
- ToString

If in the future, we need to except some fields we can add the fields which needs to be added.

davvd commented:
@sebing Thanks for your pull request, let me find someone who can review it

davvd commented:
@yegor256 review it please, it is a rather short pull request, no need to do a full code review

yegor256 commented:
@sebing your assumption is wrong. Lombok doesn't include `transient` fields by default. try to create a test, to validate that these fields are included and you will see that they are not :( (unless my information about Lombok is outdated)

sebing commented:
@yegor256 I have added a test to check the equals. 
- It is failing without my changes. 
- I have added a puzzle for correcting the equals. 

PS: Sorry for the delay in response as I am in my holiday.

sebing commented:
@yegor256 You are right about the transient variable ;)
I have added a test to check the equals. 
- It is failing without my changes too.
- I have added a puzzle for correcting the equals. 
- EqualsAndHashCode is failing without the + string, for the 'static-access' '-werror' (see the previous failed build)

PS: Sorry for the delay in answering.

yegor256 commented on src/test/java/org/takes/facets/fork/TkForkTest.java:
```diff
@@ -38,6 +42,10 @@
+    /**
+     * Regular expression constant for FkRegex creation.
+     */
+    private static final String PATTERN = "/h[a-z]{2}";
```
@sebing see http://www.yegor256.com/2016/05/03/test-methods-must-share-nothing.html

sebing replied:
@yegor256 Fixed the testing issue.

yegor256 commented on src/main/java/org/takes/facets/fork/TkFork.java:
```diff
@@ -63,11 +63,17 @@
-@ToString(of = "forks")
-@EqualsAndHashCode(of = "forks")
+@ToString(of = TkFork.LOMBOK)
+@EqualsAndHashCode(of = TkFork.LOMBOK + "")
```
@sebing what this `+ ""` is for? :)

sebing replied:
@yegor256 I am ashamed to say, i have no idea. :)
The build is failing with the compiler warning "static-access" without that.

yegor256 commented on src/main/java/org/takes/facets/fork/TkFork.java:
```diff
@@ -63,11 +63,17 @@
+     * Constant for the Lombok.
+     * Lombok doesn't include transient fields by default.
+     */
+    static final String LOMBOK = "forks";
```
@sebing I don't like this idea at all...

sebing replied:
@yegor256 I am not sure, what is the right design.
We need the information in the pre-compiler as Annotations are handled in compile time. We cannot pass objects to annotations in my understanding.

sebing replied:
@yegor256 I have removed this code, as the time is passed for this task.

yegor256 commented:
@sebing see some comments above

sebing commented:
@yegor256 I have removed the Lombok code refactoring as the time passed for fixing the issue.

I still feel my test changes are valid so you can accept the merge?

yegor256 commented:
@rultor try to merge

rultor commented:
> @rultor try to merge

@yegor256 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/9201-236998325)

rultor commented:
> @rultor try to merge

@sebing @yegor256 Oops, I failed. You can see the full log [here](http://www.rultor.com/t/9201-236998325) (spent 2min)

```
+ set -e
+ set -o pipefail
++ dirname ./run.sh
+ cd .
+ echo 17548
+ echo '2.0-SNAPSHOT BUILD'
2.0-SNAPSHOT BUILD
+ date
Tue Aug  2 18:33:47 UTC 2016
+ uptime
 18:33:47 up 14 days, 22:05,  0 users,  load average: 2.04, 4.66, 4.87
+ fork=git@github.com:sebing/takes.git
+ fork_branch=issues.697
+ head=git@github.com:yegor256/takes.git
+ head_branch=master
+ pull_id=700
+ pull_title='fixing lombok constants'
+ vars=('--env=MAVEN_OPTS=-XX:MaxPermSize=256m -Xmx1g' '--env=JAVA_OPTS=-XX:MaxPermSize=256m -Xmx1g' '--env=fork=git@github.com:sebing/takes.git' '--env=fork_branch=issues.697' '--env=head=git@github.com:yegor256/takes.git' '--env=head_branch=master' '--env=pull_id=700' '--env=pull_title=fixing lombok constants')
+ image=yegor256/rultor
+ directory=
+ scripts=('sudo locale-gen en_US en_US.UTF-8' ';' 'sudo dpkg-reconfigure locales' ';' 'export LC_ALL=en_US.UTF-8' ';' 'export LANG=en_US.UTF-8' ';' 'export LANGUAGE=en_US.UTF-8' ';' 'sudo gem install --no-rdoc --no-ri pdd' ';' 'sudo gem install --no-rdoc --no-ri est' ';' 'version=$(curl -K ../curl-appveyor.cfg --data "{accountName: '\''yegor256'\'', projectSlug: '\''takes'\'', pullRequestId: '\''${pull_id}'\''}" https://ci.appveyor.com/api/builds | jq -r '\''.version'\'')' ';' 'while true' ';' 'do status=$(curl -K ../curl-appveyor.cfg https://ci.appveyor.com/api/projects/yegor256/takes/build/${version} | jq -r '\''.build.status'\'')' ';' 'if [ "${status}" == "null" ]; then exit 1; fi' ';' 'if [ "${status}" == "success" ]; then break; fi' ';' 'if [ "${status}" == "failed" ]; then echo "see https://ci.appveyor.com/project/yegor256/takes/build/${version}"; exit 1; fi' ';' 'echo "waiting for AppVeyor build ${version}: ${status}"' ';' 'sleep 5s' ';' 'done' ';' 'mvn clean install -Ptakes -Pqulice --errors --settings ../settings.xml' ';' 'mvn clean site -Psite --errors --settings ../settings.xml' ';' 'mvn clean' ';' 'pdd --source=$(pwd) --verbose --file=/dev/null' ';')
+ squash=false
+ ff=default
+ rebase=false
+ container=yegor256_takes_700
+ as_root=false
+ mkdir -p /home/rultor/.ssh
+ echo -e 'Host github.com\n\tStrictHostKeyChecking no\n'
+ chmod 600 /home/rultor/.ssh/config
+ git clone git@github.com:yegor256/takes.git repo
Cloning into 'repo'...
+ cd repo
+ git config user.email me@rultor.com
+ git config user.name rultor
+ '[' -z 'sudo locale-gen en_US en_US.UTF-8' ']'
+ cd ..
+ cat
+ '[' false = true ']'
+ cat
+ chmod a+x entry.sh
+ cat
+ echo 'sudo locale-gen en_US en_US.UTF-8' ';' 'sudo dpkg-reconfigure locales' ';' 'export LC_ALL=en_US.UTF-8' ';' 'export LANG=en_US.UTF-8' ';' 'export LANGUAGE=en_US.UTF-8' ';' 'sudo gem install --no-rdoc --no-ri pdd' ';' 'sudo gem install --no-rdoc --no-ri est' ';' 'version=$(curl -K ../curl-appveyor.cfg --data "{accountName: '\''yegor256'\'', projectSlug: '\''takes'\'', pullRequestId: '\''${pull_id}'\''}" https://ci.appveyor.com/api/builds | jq -r '\''.version'\'')' ';' 'while true' ';' 'do status=$(curl -K ../curl-appveyor.cfg https://ci.appveyor.com/api/projects/yegor256/takes/build/${version} | jq -r '\''.build.status'\'')' ';' 'if [ "${status}" == "null" ]; then exit 1; fi' ';' 'if [ "${status}" == "success" ]; then break; fi' ';' 'if [ "${status}" == "failed" ]; then echo "see https://ci.appveyor.com/project/yegor256/takes/build/${version}"; exit 1; fi' ';' 'echo "waiting for AppVeyor build ${version}: ${status}"' ';' 'sleep 5s' ';' done ';' 'mvn clean install -Ptakes -Pqulice --errors --settings ../settings.xml' ';' 'mvn clean site -Psite --errors --settings ../settings.xml' ';' 'mvn clean' ';' 'pdd --source=$(pwd) --verbose --file=/dev/null' ';'
+ gpgconf --reload gpg-agent
+ gpg --keyserver hkp://pool.sks-keyservers.net --verbose --recv-keys 9AF0FA4C
gpg: requesting key 9AF0FA4C from hkp server pool.sks-keyservers.net
gpgkeys: key 9AF0FA4C can't be retrieved
gpg: no valid OpenPGP data found.
gpg: Total number processed: 0
'cid' file is absent, container wasn't started correctly

```

sebing commented:
@yegor256 do you think we can merge it again?

yegor256 commented:
@rultor try to merge

rultor commented:
> @rultor try to merge

@yegor256 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/9201-238325812)

rultor commented:
> @rultor try to merge

@sebing @yegor256 Oops, I failed. You can see the full log [here](http://www.rultor.com/t/9201-238325812) (spent 26min)

```
INFO: "reading src/test/java/org/takes/rq/package-info.java..."
INFO: "reading src/test/java/org/takes/rs/BodyTest.java..."
INFO: "reading src/test/java/org/takes/rs/RsGzipTest.java..."
INFO: "reading src/test/java/org/takes/rs/RsJsonTest.java..."
INFO: "reading src/test/java/org/takes/rs/RsPrettyJsonTest.java..."
INFO: "reading src/test/java/org/takes/rs/RsPrettyXmlTest.java..."
INFO: "reading src/test/java/org/takes/rs/RsPrintTest.java..."
INFO: "reading src/test/java/org/takes/rs/RsTextTest.java..."
INFO: "reading src/test/java/org/takes/rs/RsVelocityTest.java..."
INFO: "reading src/test/java/org/takes/rs/RsWithCookieTest.java..."
INFO: "reading src/test/java/org/takes/rs/RsWithHeaderTest.java..."
INFO: "reading src/test/java/org/takes/rs/RsWithHeadersTest.java..."
INFO: "reading src/test/java/org/takes/rs/RsWithStatusTest.java..."
INFO: "reading src/test/java/org/takes/rs/RsWithTypeTest.java..."
INFO: "reading src/test/java/org/takes/rs/RsWithoutHeaderTest.java..."
INFO: "reading src/test/java/org/takes/rs/RsXsltTest.java..."
INFO: "reading src/test/java/org/takes/rs/package-info.java..."
INFO: "reading src/test/java/org/takes/rs/xe/RsXemblyTest.java..."
INFO: "reading src/test/java/org/takes/rs/xe/XeAppendTest.java..."
INFO: "reading src/test/java/org/takes/rs/xe/XeDateTest.java..."
INFO: "reading src/test/java/org/takes/rs/xe/XeLinkHomeTest.java..."
INFO: "reading src/test/java/org/takes/rs/xe/XeLinkSelfTest.java..."
INFO: "reading src/test/java/org/takes/rs/xe/XeLinkTest.java..."
INFO: "reading src/test/java/org/takes/rs/xe/XeLocalhostTest.java..."
INFO: "reading src/test/java/org/takes/rs/xe/XeSlaTest.java..."
INFO: "reading src/test/java/org/takes/rs/xe/XeTransformTest.java..."
INFO: "reading src/test/java/org/takes/rs/xe/XeWhenTest.java..."
INFO: "reading src/test/java/org/takes/rs/xe/package-info.java..."
INFO: "reading src/test/java/org/takes/tk/TkClasspathTest.java..."
INFO: "reading src/test/java/org/takes/tk/TkCorsTest.java..."
INFO: "reading src/test/java/org/takes/tk/TkFilesTest.java..."
INFO: "reading src/test/java/org/takes/tk/TkGzipTest.java..."
INFO: "reading src/test/java/org/takes/tk/TkHtmlTest.java..."
INFO: "reading src/test/java/org/takes/tk/TkMeasuredTest.java..."
INFO: "reading src/test/java/org/takes/tk/TkProxyTest.java..."
INFO: "reading src/test/java/org/takes/tk/TkReadAlwaysTest.java..."
INFO: "reading src/test/java/org/takes/tk/TkRedirectTest.java..."
INFO: "reading src/test/java/org/takes/tk/TkRetryTest.java..."
INFO: "reading src/test/java/org/takes/tk/TkSlf4jTest.java..."
INFO: "reading src/test/java/org/takes/tk/TkTextTest.java..."
INFO: "reading src/test/java/org/takes/tk/TkVerboseTest.java..."
INFO: "reading src/test/java/org/takes/tk/TkVersionedTest.java..."
INFO: "reading src/test/java/org/takes/tk/TkWithHeadersTest.java..."
INFO: "reading src/test/java/org/takes/tk/package-info.java..."
INFO: "reading src/test/resources/log4j.properties..."
INFO: "reading src/test/resources/org/takes/rs/inclass/imported.xsl..."
INFO: "reading src/test/resources/org/takes/rs/included.xsl..."
INFO: "reading src/test/resources/org/takes/rs/includes.xsl..."
INFO: "reading src/test/resources/org/takes/rs/simple.xsl..."
INFO: "reading src/verifier/verifications.xml..."
INFO: "reading src/www.takes.org/index.html..."
ERROR: "puzzle src/test/java/org/takes/facets/fork/TkForkTest.java:73-74 has a very short description of just 14 words while a minimum of 20 is required"
/var/lib/gems/2.2.0/gems/pdd-0.15.4/lib/pdd.rb:150:in `rules': 1 errors, see log above (PDD::Error)
	from /var/lib/gems/2.2.0/gems/pdd-0.15.4/lib/pdd.rb:92:in `xml'
	from /var/lib/gems/2.2.0/gems/pdd-0.15.4/bin/pdd:90:in `<top (required)>'
	from /usr/local/bin/pdd:23:in `load'
	from /usr/local/bin/pdd:23:in `<main>'
container 41d1b37ad78d704b7ecea827d9a31d20ed9707dd3bf9d3568fa1cb076604d355 is dead
Mon Aug  8 18:36:36 UTC 2016

```

yegor256 commented:
@rultor try to merge

rultor commented:
> @rultor try to merge

@yegor256 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/9201-307029786)

rultor commented:
> @rultor try to merge

@sebing @yegor256 Oops, I failed. You can see the full log [here](http://www.rultor.com/t/9201-307029786) (spent 8min)

```
+ status=running
+ '[' running == null ']'
+ '[' running == success ']'
+ '[' running == failed ']'
+ echo 'waiting for AppVeyor build 2826: running'
waiting for AppVeyor build 2826: running
+ sleep 5s
+ true
++ curl -K ../curl-appveyor.cfg https://ci.appveyor.com/api/projects/yegor256/takes/build/2826
++ jq -r .build.status
+ status=running
+ '[' running == null ']'
+ '[' running == success ']'
+ '[' running == failed ']'
+ echo 'waiting for AppVeyor build 2826: running'
waiting for AppVeyor build 2826: running
+ sleep 5s
+ true
++ curl -K ../curl-appveyor.cfg https://ci.appveyor.com/api/projects/yegor256/takes/build/2826
++ jq -r .build.status
+ status=running
+ '[' running == null ']'
+ '[' running == success ']'
+ '[' running == failed ']'
+ echo 'waiting for AppVeyor build 2826: running'
waiting for AppVeyor build 2826: running
+ sleep 5s
+ true
++ curl -K ../curl-appveyor.cfg https://ci.appveyor.com/api/projects/yegor256/takes/build/2826
++ jq -r .build.status
+ status=running
+ '[' running == null ']'
+ '[' running == success ']'
+ '[' running == failed ']'
+ echo 'waiting for AppVeyor build 2826: running'
waiting for AppVeyor build 2826: running
+ sleep 5s
+ true
++ curl -K ../curl-appveyor.cfg https://ci.appveyor.com/api/projects/yegor256/takes/build/2826
++ jq -r .build.status
+ status=running
+ '[' running == null ']'
+ '[' running == success ']'
+ '[' running == failed ']'
+ echo 'waiting for AppVeyor build 2826: running'
waiting for AppVeyor build 2826: running
+ sleep 5s
+ true
++ curl -K ../curl-appveyor.cfg https://ci.appveyor.com/api/projects/yegor256/takes/build/2826
++ jq -r .build.status
+ status=failed
+ '[' failed == null ']'
+ '[' failed == success ']'
+ '[' failed == failed ']'
+ echo 'see https://ci.appveyor.com/project/yegor256/takes/build/2826'
see https://ci.appveyor.com/project/yegor256/takes/build/2826
+ exit 1
container 8e94c5cfd7e0c626f4d8f95c78d03e3abddd4a4e16bc037f515bf7810d91ef7b is dead
Thu Jun  8 10:31:57 CEST 2017

```

paulodamaso commented:
@sebing build is failing, please take a look

paulodamaso commented:
@sebing just checking, thanks

paulodamaso commented:
@0crat out

0crat commented:
> @0crat out ([here](https://github.com/yegor256/takes/issues/700#issuecomment-432725219))

@paulodamaso Job `gh:yegor256/takes#700` is not assigned, can't get performer

<!-- https://www.0crat.com/footprint/C3FFK3YAY/4ec422b8-6d07-40ff-920f-a038b3c0d68e, version: 0.37.4, hash: ${buildNumber} -->

0crat commented:
This job is not in scope

<!-- https://www.0crat.com/footprint/C3FFK3YAY/29a478b0-cfe2-4399-a06f-7b710b72775c, version: 0.37.4, hash: ${buildNumber} -->

0crat commented:
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@1f655e82-263b-4d1e-9275-83d5854cf304, thread: AsyncProc-4
com.jcabi.xml.StrictXML[124] java.lang.IllegalArgumentException: 1 error(s) in XML document: -1:-1: Duplicate unique value [gh:yegor256/takes#700] declared for identity constraint of element "blanks".
```

[0.38.5](https://github.com/zerocracy/farm): CID: [288f84ff-5cf7-4f71-9bf3-b3e5a1d9873f](https://www.0crat.com/footprint/C3FFK3YAY/288f84ff-5cf7-4f71-9bf3-b3e5a1d9873f), Type: "Close job"

<!-- https://www.0crat.com/footprint/C3FFK3YAY/3cd9c2a1-c90e-4ce2-82c1-3cc73092db0a, version: 0.38.5, hash: ${buildNumber} -->

0crat commented:
Job `gh:yegor256/takes#700` is not assigned, can't get performer

<!-- https://www.0crat.com/footprint/C3FFK3YAY/cc0b67d3-56aa-4680-89a5-3ad6e0e72c72, version: 0.38.5, hash: ${buildNumber} -->
